File: QuickStart/TreeView/TreeExamplesControl.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports Telerik.QuickStart.WinControls
Imports Telerik.WinControls.UI

Namespace Telerik.Examples
	Public Class TreeExamplesControl
		Inherits ExamplesForm
		Private topRightBitmap As Bitmap = My.Resources.TV_gradient
		Private headerPanel As TreeExampleHeaderPanel

		Public Sub New()
			headerPanel = New TreeExampleHeaderPanel()
			Me.Controls.Add(headerPanel)
			Me.AutoScroll = False
		End Sub

		Public Overridable ReadOnly Property TreeControl() As RadTreeView
			Get
				Return Nothing
			End Get
		End Property

		Public Overridable ReadOnly Property BottomRightImage() As Bitmap
			Get
				Return Nothing
			End Get
		End Property

		Public Overridable ReadOnly Property HeaderText() As String
			Get
				Return ""
			End Get
		End Property

		Public Overridable ReadOnly Property TopRightImage() As Bitmap
			Get
				Return Me.topRightBitmap
			End Get
		End Property

		Protected Overridable Function CalculateTreeWidth(ByVal clientWidth As Integer) As Integer
			Return clientWidth - (197 + 264)
		End Function

		Protected Overridable Function CalculateOffset(ByVal clientWidth As Integer) As Integer
			Return 194
		End Function

		Protected Overridable Sub UpdateTreeDimensions()
			If Me.TreeControl IsNot Nothing Then
				Dim width As Integer = Me.ClientSize.Width
				Dim height As Integer = Me.ClientSize.Height
				Dim offset As Integer = CalculateOffset(width)
				Me.TreeControl.Size = New Size(CalculateTreeWidth(width), height - 94 - 30)
				Me.TreeControl.Location = New Point(offset, 62 + 30)
				Me.headerPanel.Size = New Size(Me.TreeControl.Size.Width, 30)
				Me.headerPanel.Location = New Point(offset, 62)
			End If
		End Sub

		Protected Overridable Sub PaintImages(ByVal g As Graphics)
			If TopRightImage IsNot Nothing Then
				g.DrawImage(TopRightImage, New Point(Me.Size.Width - TopRightImage.Width, 0))
			End If
			If BottomRightImage IsNot Nothing Then
				g.DrawImage(BottomRightImage, New Point(Me.Size.Width - BottomRightImage.Width, Me.Size.Height - BottomRightImage.Height))
			End If
		End Sub

		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)
			If TreeControl IsNot Nothing Then
				Me.TreeControl.ItemHeight = 27
				Me.TreeControl.ShowLines = False
				Me.TreeControl.FullRowSelect = True
				Me.TreeControl.ExpandAnimation = ExpandAnimation.None
				Me.headerPanel.Text = HeaderText
				Me.TreeControl.AllowEdit = True
				Me.TreeControl.AllowAdd = False
				Me.TreeControl.AllowRemove = False
				Me.TreeControl.AllowDefaultContextMenu = True
			End If
		End Sub

		Protected Overrides Sub OnSizeChanged(ByVal e As EventArgs)
			MyBase.OnSizeChanged(e)
			UpdateTreeDimensions()
		End Sub

		Protected Overrides Sub OnPaint(ByVal e As PaintEventArgs)
			MyBase.OnPaint(e)
			PaintImages(e.Graphics)
		End Sub
	End Class
End Namespace

File: QuickStart/TreeView/TreeExamplesControl.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Telerik.Examples.WinControls.Properties;
using Telerik.QuickStart.WinControls;
using Telerik.WinControls.UI;

namespace Telerik.Examples
{
    public class TreeExamplesControl : ExamplesForm
    {
        Bitmap topRightBitmap = Resources.TV_gradient;
        TreeExampleHeaderPanel headerPanel;

        public TreeExamplesControl()
        {
            headerPanel = new TreeExampleHeaderPanel();
            this.Controls.Add(headerPanel);
            this.AutoScroll = false;
        }

        public virtual RadTreeView TreeControl { get { return null; } }

        public virtual Bitmap BottomRightImage { get { return null; } }

        public virtual string HeaderText { get { return ""; } }

        public virtual Bitmap TopRightImage
        {
            get { return this.topRightBitmap; }
        }

        protected virtual int CalculateTreeWidth(int clientWidth)
        {
            return clientWidth - (197 + 264);
        }

        protected virtual int CalculateOffset(int clientWidth)
        {
            return 194;
        }

        protected virtual void UpdateTreeDimensions()
        {
            if (this.TreeControl != null)
            {
                int width = this.ClientSize.Width;
                int height = this.ClientSize.Height;
                int offset = CalculateOffset(width);
                this.TreeControl.Size = new Size(CalculateTreeWidth(width), height - 94 - 30);
                this.TreeControl.Location = new Point(offset, 62 + 30);
                this.headerPanel.Size = new Size(this.TreeControl.Size.Width, 30);
                this.headerPanel.Location = new Point(offset, 62);
            }
        }

        protected virtual void PaintImages(Graphics g)
        {
            if (TopRightImage != null)
            {
                g.DrawImage(TopRightImage, new Point(this.Size.Width - TopRightImage.Width, 0));
            }
            if (BottomRightImage != null)
            {
                g.DrawImage(BottomRightImage, new Point(this.Size.Width - BottomRightImage.Width, this.Size.Height - BottomRightImage.Height));
            }
        }

        protected override void OnLoad(EventArgs e)
        {
            base.OnLoad(e);
            if (TreeControl != null)
            {
                this.TreeControl.ItemHeight = 27;
                this.TreeControl.ShowLines = false;
                this.TreeControl.FullRowSelect = true;
                this.TreeControl.ExpandAnimation = ExpandAnimation.None;
                this.headerPanel.Text = HeaderText;
                this.TreeControl.AllowEdit = true;
                this.TreeControl.AllowAdd = false;
                this.TreeControl.AllowRemove = false;
                this.TreeControl.AllowDefaultContextMenu = true;
            }
        }

        protected override void OnSizeChanged(EventArgs e)
        {
            base.OnSizeChanged(e);
            UpdateTreeDimensions();
        }

        protected override void OnPaint(PaintEventArgs e)
        {
            base.OnPaint(e);
            PaintImages(e.Graphics);
        }
    }
}
